test/integration/user/test/interactivity.test.js

```diff
@@ -197,89 +197,114 @@ describe('Interactivity', () => {
                 });
             });
         });
+    });
+
+    describe('when the user move the mouse on the map', () => {
+        describe('and the mouse enters in a feature', () => {
+            it('should fire a featureHover event with a features list containing the entered feature', done => {
+                interactivity.on('featureHover', event => {
+                    expect(event.features[0].id).toEqual(-0);
+                    expect(event.features[0].layerId).toEqual('layer1');
+                    done();
+                });
+                onLoaded(() => {
+                    // Move mouse inside a feature 1
+                    util.simulateMove({ lng: 5, lat: 5 });
+                });
+            });
+
+            it('should fire a featureEnter event with a features list containing the entered feature', done => {
+                interactivity.on('featureEnter', event => {
+                    expect(event.features[0].id).toEqual(-0);
+                    expect(event.features[0].layerId).toEqual('layer1');
+                    done();
+                });
+                onLoaded(() => {
+                    // Move mouse inside a feature 1
+                    util.simulateMove({ lng: 5, lat: 5 });
+                });
+            });
+
+            it('should not fire a featureEnter event when the mouse is moved inside the same feature', done => {
+                const featureEnterSpy = jasmine.createSpy('featureEnterSpy');
+                onLoaded(() => {
+                    interactivity.on('featureEnter', featureEnterSpy);
+                    // Move mouse inside a feature 1
+                    util.simulateMove({ lng: 5, lat: 5 });
+                    // Move mouse inside the same feature 1
+                    util.simulateMove({ lng: 5, lat: 15 });
+                    setTimeout(() => {
+                        expect(featureEnterSpy).toHaveBeenCalledTimes(1);
+                        done();
+                    }, 0);
+                });
+            });
+        });
 
-        describe('when the user move the mouse on the map', () => {
-            describe('and the mouse enters in a feature', () => {
-                it('should fire a featureHover event with a features list containing the entered feature', done => {
+        describe('and the mouse leaves a feature', () => {
+            it('should fire a featureHover event with an empty features list', done => {
```

Comment:
Why are we rewriting these tests?

Reply:
https://github.com/CartoDB/carto-vl/pull/865#discussion_r211219967